doc/source/whatsnew/v2.2.0.rst

```diff
@@ -29,7 +29,7 @@ enhancement2
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
 - DataFrame.apply now allows the usage of numba (via ``engine="numba"``) to JIT compile the passed function, allowing for potential speedups (:issue:`54666`)
--
+- It is now possible to pass an instance of a ``collections.UserDict`` subclass to initialize a DataFrame. The expected behaviour is the same as if you pass a ``dict`` (:issue:`55109`)
```

Comment:
Maybe, would it be better to rephrase that to include the explicit reference to ``Mapping`` instead of giving an example of the usage as ``UserDict`` ?
```suggestion
- It is now possible to pass pass an object that either directly inherits from ``abc.Mapping`` or has it in its inheritance chain to initialize a DataFrame. The expected behaviour is the same as if you pass a ``dict`` object. (:issue:`55109`)
```